### PR TITLE
Bug 1863000: deployment of vmware fails if URL contains space

### PR DESCRIPTION
**Pull request by glekner**

### CODE REVIEW

**Comment by glekner** [comment]
@suomiy thoughts?

File: frontend/packages/kubevirt-plugin/src/components/create-vm-wizard/redux/validations/providers/v2vvmware-tab-validation.ts [hunk 1]
```diff
@@ -18,14 +18,25 @@ export const getV2VVMwareImportProvidersTabValidity = (
 
   const hasSelectedVM = !!iGetVMWareFieldValue(state, id, VMWareProviderField.VM);
   const hasLoadedVM = !!iGetVMWareFieldAttribute(state, id, VMWareProviderField.VM, 'vm');
+  // check if host name has any kind of whitespace
+  const isHostnameValid = !/\s/.test(iGetVMWareFieldValue(state, id, VMWareProviderField.HOSTNAME));
```
Comment (yaacov) [inline frontend/packages/kubevirt-plugin/src/components/create-vm-wizard/redux/validations/providers/v2vvmware-tab-validation.ts:22]: the BZ is just about spaces at the end, am I missing something ?

Comment (glekner) [reply frontend/packages/kubevirt-plugin/src/components/create-vm-wizard/redux/validations/providers/v2vvmware-tab-validation.ts:22]: why not check for any spaces while we're here? :)

Comment (yaacov) [reply frontend/packages/kubevirt-plugin/src/components/create-vm-wizard/redux/validations/providers/v2vvmware-tab-validation.ts:22]: can we truncate the hostname ?

File: frontend/packages/kubevirt-plugin/src/components/create-vm-wizard/redux/validations/providers/v2vvmware-tab-validation.ts [hunk 2]
```diff
@@ -18,14 +18,25 @@ export const getV2VVMwareImportProvidersTabValidity = (
 
   const hasSelectedVM = !!iGetVMWareFieldValue(state, id, VMWareProviderField.VM);
   const hasLoadedVM = !!iGetVMWareFieldAttribute(state, id, VMWareProviderField.VM, 'vm');
+  // check if host name has any kind of whitespace
```
Comment (suomiy) [inline frontend/packages/kubevirt-plugin/src/components/create-vm-wizard/redux/validations/providers/v2vvmware-tab-validation.ts:21]: this logic should be in a new function called validateV2VVMwareImportProvidersTab. 
We could also keep the in validationConfig format as a convention (you can take an inspiration from `ovirt-provider-tab-validation`)

Comment (suomiy) [reply frontend/packages/kubevirt-plugin/src/components/create-vm-wizard/redux/validations/providers/v2vvmware-tab-validation.ts:21]: also the check and save button should reflect the error state of this validation

**Comment by glekner** [comment]
Screen Shot 2020-08-11 at 11 52 17

@suomiy 
Added `validateVMwareSettings` function.
also added disable for Check and save button in case of invalid input.

File: frontend/packages/kubevirt-plugin/src/components/create-vm-wizard/tabs/import-providers-tab/providers/vm-import-password.tsx [hunk 3]
```diff
@@ -96,6 +97,12 @@ const stateToProps = (state, { wizardReduxID, provider }) => {
     );
   }
 
+  if (provider === VMImportProvider.VMWARE && hasAllPrerequisiteValuesFiled) {
```
Comment (suomiy) [inline frontend/packages/kubevirt-plugin/src/components/create-vm-wizard/tabs/import-providers-tab/providers/vm-import-password.tsx:100]: we don't have to do the test second time. You can just check if the validation is Error type.And you can do this for both providers with iGetProviderFieldAttribute

**Comment by suomiy** [comment]
> Bug 1863000 should be left for the controller only.

let this bug for the UI and created a new one for the controller https://bugzilla.redhat.com/show_bug.cgi?id=1867980

File: frontend/packages/kubevirt-plugin/src/components/create-vm-wizard/tabs/import-providers-tab/providers/vm-import-password.tsx [hunk 4]
```diff
@@ -86,6 +87,14 @@ const stateToProps = (state, { wizardReduxID, provider }) => {
       provider,
       OvirtProviderField.USERNAME,
       VMWareProviderField.USERNAME,
+    ) &&
+    !iGetProviderFieldAttribute(
```
Comment (suomiy) [inline frontend/packages/kubevirt-plugin/src/components/create-vm-wizard/tabs/import-providers-tab/providers/vm-import-password.tsx:91]: we have to check the validation type. It should fail for error/trivial error but pass for warn/info

File: frontend/packages/kubevirt-plugin/src/components/create-vm-wizard/tabs/import-providers-tab/providers/vm-import-password.tsx [hunk 5]
```diff
@@ -70,6 +72,17 @@ const stateToProps = (state, { wizardReduxID, provider }) => {
...
+  const validationField = iGetProviderFieldAttribute(
+    state,
+    wizardReduxID,
+    provider,
+    'validation',
+    OvirtProviderField.API_URL,
+    VMWareProviderField.HOSTNAME,
+  );
+  const isValidationCritical =
+    iGet(validationField, 'type') === ValidationErrorType.Error ||
```
Comment (suomiy) [inline frontend/packages/kubevirt-plugin/src/components/create-vm-wizard/tabs/import-providers-tab/providers/vm-import-password.tsx:84]: let's use `[].includes` here for brevity

Comment (glekner) [reply frontend/packages/kubevirt-plugin/src/components/create-vm-wizard/tabs/import-providers-tab/providers/vm-import-password.tsx]: fixed

**Pull request merged**